### PR TITLE
CI test fixes and improvements

### PR DESCRIPTION
**Pull request by rajivshah3**
- Align CI test with [`sccache` caveats](https://github.com/mozilla/sccache#rust)
  - Create lib crate instead of bin crate — `sccache` cannot cache bin crates 
  - Set CARGO_INCREMENTAL=0 as required by sccache
- Check cache hits or cache misses > 0 to make sure `sccache` is working

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -51,5 +51,18 @@ jobs:
...
-    - run: cargo new project && cd project && cargo build
-    # todo, add that it actually used it, maybe by show stats
\ No newline at end of file
+    - name: Create and build test project
+      run: |
+        cargo new project --lib
+        cd project
+        cargo build
+      env:
+        CARGO_INCREMENTAL: 0
```
Comment (rajivshah3) [inline .github/workflows/ci.yaml:62]: Since we set `RUSTC_WRAPPER` as part of the action, should we be setting `CARGO_INCREMENTAL` as well?

Comment (aviramha) [reply .github/workflows/ci.yaml:60]: I don't think so, atleast not by default.

Comment (aviramha) [reply .github/workflows/ci.yaml:62]: Nice catch! I think if it's required for scache to work that makes sense to set

**Comment by aviramha** [comment]
Please resolve conflicts for merge :)

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (rajivshah3) [reply .github/workflows/ci.yaml:60]: Ok! Will add that in a separate PR

Comment (rajivshah3) [reply .github/workflows/ci.yaml:60]: Ok! I'll add that in a separate PR

Comment (rajivshah3) [reply .github/workflows/ci.yaml:62]: Ok! I'll add that in a separate PR

**Comment by rajivshah3** [comment]
> /home/runner/work/_temp/8f604a8b-73e5-49a0-9706-1a23e046333b.sh: line 2: [: null: integer expression expected

Hmm, I'll have to check on why this is happening

**Comment by rajivshah3** [comment]
I made a few mistakes and forgot that checking the cache size is not implemented when using the GHA cache. I think checking to make sure either the cache hits or cache misses are nonzero should work, since that indicates that the cache was accessed

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (rajivshah3) [reply .github/workflows/ci.yaml:62]: Opened https://github.com/metalbear-co/sccache-action/pull/9

**Review by aviramha — APPROVED** [review]

**Pull request merged**